lib/tenstorrent/bh_arc/aiclk_ppm.c

```diff
@@ -64,6 +64,12 @@ void CalculateTargAiclk(void)
 	/* (it will not be above Fmax, since we calculated the max limits last) */
 	aiclk_ppm.targ_freq = MAX(targ_freq, aiclk_ppm.fmin);
 
+	/* Apply random frequency if sweep is enabled */
+	if (aiclk_ppm.sweep_en == 1) {
+		aiclk_ppm.targ_freq = rand() % (aiclk_ppm.sweep_high - aiclk_ppm.sweep_low + 1)
+							  + aiclk_ppm.sweep_low;
+	}
```

Comment:
This is generally not great code to have enabled in production. Consider the case where a stack overflow occurs and some rogue memory write flips `sweep_en` to 1. Bad things could happen 😬